### ISSUE TITLE
Make rules

### ISSUE THREAD

**Issue by brettin** [JDACS4C-IMPROVE/Singularity#5]
Rule default: if the def file is newer than the sif file, build the sif file from the def file

Rule make deploy: default + cp all sif files to prefix/images

**Comment by brettin** [comment]
From Andreas in Slack: initial Makefile is done but i did run into some issues we should briefly address with Oleksandr and you. I made some assumption from where to build the container and what is in the search path but aside some config it seems to work. 

[11:09 (https://improve-smw1216.slack.com/archives/D046XECHAQ4/p1668575352657279)
e.g.
sh src/bootstrap.sh -h 
src/bootstrap.sh: line 6: ../config/improve.env: No such file or directory

**Comment by brettin** [comment]
Either change the current working directory to src inside the make file, or let's set the location of the config dir in the make file and pass it as a param to bootstrap.sh

**Comment by wilke** [comment]
@brettin Some questions :

1. how are we building singularity images, is bootstrap.sh the only blessed method?
2. What is the build context for the container or more precisely what is the "root" directory to build the container from. Do we assume everything is relative to ./Singularity (preferred) or ./Singularity/definitions or ./Singularity/src
3. Is bootstrap.sh a free standing tool and should be executable and in the $PATH and should it be part of the install/setup process (install.sh)

**Comment by brettin** [comment]
1. we should have one blessed method so that they are all built/named the same way.
2. ./Singularity
3. it is a tool to help build containers in a reproducible manner.

This might solve your problem:
https://stackoverflow.com/questions/1789594/how-do-i-write-the-cd-command-in-a-makefile

**Comment by brettin** [comment]
I added a -s flag to bootstrap.sh. If this is specified, then the user is NOT automatically logged into a sandbox. While the natural usage would be if -s is specified, a user is logged into a sandbox, but I didn't want to change the default behavior.

I also took care of the hard-coded paths in bootstrap.sh

```
SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
source ${SCRIPT_DIR}"/../config/improve.env"
source ${SCRIPT_DIR}"/login.sh" "$ISL/${NAME}-${IMAGE}-${DATE}"
```

The remaining issue that I see is this hard-coded path: `../src/Singularity_gpu_fix.sh` that exists in our definition files.
